src/core/qgsdistancearea.cpp

```diff
@@ -537,6 +540,111 @@ double QgsDistanceArea::measureLine( const QgsPoint &p1, const QgsPoint &p2, Qgs
   return result;
 }
 
+double QgsDistanceArea::measureLineProjected( const QgsPoint &p1, double distance, double azimuth, QgsPoint *projected_point )
+{
+  double result = 0.0;
+  QgsPoint p2;
+  if ( geographic() )
+  {
+    if ( !mEllipsoidalMode )
+    {
+      if ( ( mSemiMajor > 0 ) && ( mSemiMinor > 0 ) )
+      {
+        setEllipsoid( mSemiMajor, mSemiMinor );
+      }
+      else
+      {
+        setEllipsoid( sourceCrs().ellipsoidAcronym() );
+      }
+      if ( mEllipsoid != GEO_NONE )
+      {
+        mEllipsoidalMode = true;
+      }
+    }
+    if ( mEllipsoid != GEO_NONE )
+    {
+      p2 = computeSpheroidProject( p1, distance, azimuth );
+      QList<QgsPoint> linePoints;
+      linePoints.append( p1 );
+      linePoints.append( p2 );
+      QgsLineString lineString = QgsLineString( linePoints );
+      result = lineString.length();
+    }
+  }
+  else // cartesian coordinates
+  {
+    result = distance; // Avoid rounding errors when using meters [return as sent]
+    if ( sourceCrs().mapUnits() != QgsUnitTypes::DistanceMeters )
+    {
+      distance = ( distance * QgsUnitTypes::fromUnitToUnitFactor( QgsUnitTypes::DistanceMeters, sourceCrs().mapUnits() ) );
+      result = measureLine( p1, p2 );
```

Comment:
Again - this is overkill for measuring cartesian distance between two points. Use `resullt = p1.distance( p2 )` instead.